### PR TITLE
Make backend code counts endline as 1 char instead of 2, like front-end

### LINKED ISSUE #4419: Server- and client-side bio and display name count endlines and emoji combinations inconsistently
Hi!

When using UTF-8 chars (multi-byte ones, I assume), there's inconsistency in the "characters left" indication. I'm using mastodon.xyz which is in 1.4.7, problem still exists in 2.5.0rc1
When I submit the edited changes, the value that comes back it's smaller than the one that it's calculated by the frontend.
* * * *

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

### PR DESCRIPTION
**Pull request by ariasuni**
Fix #4419

### CODE REVIEW

**Review by danhunsaker — APPROVED** [review]
This is the way I was going to suggest doing it if you hadn't already, so +100!

File: app/models/account.rb [hunk 1]
```diff
@@ -112,6 +112,11 @@ class Account < ApplicationRecord
 
   delegate :chosen_languages, to: :user, prefix: false, allow_nil: true
 
+  # Convert newlines so validation doesn't count them as two characters
+  def note=(val)
+    self['note'] = val.gsub("\r\n", "\n")
```
Comment (Gargron) [inline app/models/account.rb]: I suggest:

```ruby
before_validation :normalize_note

private

def normalize_note
  self.note = note.gsub("\r\n", "\n")
end
```

Worth noting I do not remember something like this for status text...

Comment (ariasuni) [reply app/models/account.rb]: This code is used for counting a status text:

https://github.com/tootsuite/mastodon/blob/bf70e5cfdaef9b7a3a927548a05203a864d5bea9/app/validators/status_length_validator.rb#L18

I couldn’t make a test triggering `before_validation`, so I didn’t go for you suggestion or for using `grapheme_length`.

**Comment by ariasuni** [comment]
Thanks.

**Pull request closed without merging**